### PR TITLE
Yaml parser added to Config-cxx. Resolves #2

### LINKED ISSUE #2: Add support for yaml config

### PR DESCRIPTION
**Pull request by srivastava-yash**
Added support for yaml parsing of configs.
Issue: #2

### CODE REVIEW

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Good, you can add logic in config.cpp to load files with yaml loader if file has yaml or yml extension

**Review by cieslarmichal — APPROVED** [review]

File: src/YamlConfigLoader.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    }
+}
+
+
+namespace
+{
+void flattenConfig(YAML::Node& configNode, std::unordered_map<std::string, ConfigValue>& configValues)
+{
+    std::function<void(const YAML::Node&, std::string, std::unordered_map<std::string, ConfigValue>&)> flatten_recursive =
```
Comment (cieslarmichal) [inline src/YamlConfigLoader.cpp]: You need `<functional>` include

File: src/YamlConfigLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    };
+    
+    flatten_recursive(configNode, "", configValues);
+}
+
+ConfigValue getScalarValue(YAML::Node node) 
+{
+     bool set = false;
+     ConfigValue node_value;
+     try
```
Comment (cieslarmichal) [inline src/YamlConfigLoader.cpp:113]: Please change all variable names to camelCase as it is everywhere in codebase

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

File: src/YamlConfigLoader.cpp (hunk 2 above)
Comment (srivastava-yash) [reply src/YamlConfigLoader.cpp:113]: resolved

File: src/YamlConfigLoader.cpp (hunk 1 above)
Comment (srivastava-yash) [reply src/YamlConfigLoader.cpp]: resolved

**Review by cieslarmichal — APPROVED** [review]

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**